runner/http.go

```diff
@@ -80,12 +81,32 @@ func (h HTTP) ID() string {
 
 // Run executes a GET request to the Path using the Client
 func (h HTTP) Run() op.Op {
+	// protect from accidental nil reference panics
+	if h.ctx == nil {
+		h.ctx = context.Background()
+	}
+
+	runCtx := h.ctx
+	var runCancelFunc context.CancelFunc
+	if h.Timeout > 0 {
+		runCtx, runCancelFunc = context.WithTimeout(h.ctx, time.Duration(h.Timeout))
+		defer runCancelFunc()
+	}
+
 	startTime := time.Now()
 
-	redactedResponse, err := h.Client.RedactGet(h.Path, h.Redactions)
+	redactedResponse, err := h.Client.RedactGetWithContext(runCtx, h.Path, h.Redactions)
 	result := map[string]any{"response": redactedResponse}
 	if err != nil {
-		op.New(h.ID(), result, op.Fail, err, Params(h), startTime, time.Now())
+		var failureType op.Status
+		if errors.Is(err, context.DeadlineExceeded) {
+			failureType = op.Timeout
+		} else if errors.Is(err, context.Canceled) {
+			failureType = op.Canceled
+		} else {
+			failureType = op.Fail
```

Comment:
I think we ought to classify these catch-all cases as `status.Unknown` and save the `op.Fail` when there's specific errors we can key off of from. Despite the overlap between "fail" and "failed but unknown why" using `Unknown` at least gives us a chance to highlight these cases.

Reply:
That makes sense. I think we've gone with `Unknown` in similar cases in other runners. Because the existing code called all errors `op.Fail`, I stuck with that behavior initially. However, I've updated the branch to go with `op.Unknown` now.